framework/test/mocha/unit/modules/chain/submodules/blocks/chain.js

```diff
@@ -591,19 +598,100 @@ describe('blocks/chain', () => {
 	});
 
 	describe('__private.applyConfirmedStep', () => {
-		/* eslint-disable mocha/no-pending-tests */
-		it('should return when block.transactions includes no transactions');
+		// Arrange
+		const filledBlock = {
+			id: 1,
+			height: 1,
+			transactions: [{ id: 21 }, { id: 22 }],
+		};
+		let originalInertTransactions;
+
+		beforeEach(async () => {
+			// Arrange (Backup)
+			originalInertTransactions = global.exceptions.inertTransactions;
+		});
+
+		afterEach(async () => {
+			// Restore
+			global.exceptions.inertTransactions = originalInertTransactions;
+		});
 
-		it('should call modules.processTransactions.undoTransactions');
+		it('should return when block.transactions includes no transactions', async () => {
+			// Arrange
+			const emptyBlock = {
+				id: 1,
+				height: 1,
+				transactions: [],
+			};
+			// Act && Assert
+			return expect(__private.applyConfirmedStep(emptyBlock)).to.eventually.be
+				.undefined;
+		});
+
+		it('should call modules.processTransactions.applyConfirmedStep', async () => {
+			// Arrange
+			global.exceptions.inertTransactions = [filledBlock.transactions[0].id];
+			const dummyTx = {};
+
+			// Act
+			await __private.applyConfirmedStep(filledBlock, dummyTx);
+
+			// Assert
+			return expect(
+				modules.processTransactions.applyTransactions
+			).to.have.been.calledWith([filledBlock.transactions[1]], dummyTx);
+		});
+
+		it('should throw error when errors exist in unappliedTransactionResponse', async () => {
+			// Arrange
+			const errors = [new Error('#Test Error')];
+			modules.processTransactions.applyTransactions.resolves({
+				...processTransactionMethodResponse,
+				transactionsResponses: [
+					{ status: TransactionStatus.OK },
+					{ status: TransactionStatus.FAIL, errors },
+				],
+			});
+
+			try {
+				// Act
+				await __private.applyConfirmedStep(filledBlock);
```

Comment:
This will give a false positive if it doesn't throw. I suggest adding an expectation to cater for this scenario like bellow:

```suggestion
				await __private.applyConfirmedStep(filledBlock);
				// Next line is to make sure the above method actually failed; should never execute.
				expect(false).to.be(true);
```

Reply:
very good catch. Thanks!